### PR TITLE
Tests: Bluetooth: fix failing unittests

### PR DESCRIPTION
**Pull request by kruithofa**
Due to some refactoring in the past a few unittests have started failing.
This PR fixes 2 tests for direction finding due to refactoring/new kconfigs
and 1 test for audio codec due to updating to BT spec 5.3

Added Piotr Pryga and Morten Priess since they were involved in the original tests
and/or code changes

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

### CODE REVIEW

**Comment by jhedberg** [comment]
Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.

**Comment by kruithofa** [comment]
> Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.

So should I prefix it with 'bluetooth: unittest' instead?

**Comment by Thalley** [comment]
> > Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.
> 
> So should I prefix it with 'bluetooth: unittest' instead?

`tests: Bluetooth` is better :)

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_EXT_ADV) && defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_EXT_ADV && CONFIG_BT_BROADCASTER */
```
Comment (Thalley) [inline subsys/bluetooth/host/hci_core.c:3803]: You sure about this? 

`bt_adv_reset_adv_pool` seems to be fine without `CONFIG_BT_EXT_ADV`

```c
void bt_adv_reset_adv_pool(void)
{
#if defined(CONFIG_BT_EXT_ADV)
	(void)memset(&adv_pool, 0, sizeof(adv_pool));
#endif /* defined(CONFIG_BT_EXT_ADV) */

	(void)memset(&bt_dev.adv, 0, sizeof(bt_dev.adv));
}
```

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/hci_core.c (hunk 1 above)
Comment (kruithofa) [reply subsys/bluetooth/host/hci_core.c:3803]: No, I discussed this with Sean and will update

**Review by sean-madigan — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 2]
```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_BROADCASTER */
```
Comment (jhedberg) [inline subsys/bluetooth/host/hci_core.c:3803]: Since this part is fixing something in the host and not a test case, wouldn't it make sense to split it into a separate commit?

**Review by jhedberg — COMMENTED** [review]
Thanks for updating the prefixes! One additional comment inline.

File: subsys/bluetooth/host/hci_core.c (hunk 2 above)
Comment (kruithofa) [reply subsys/bluetooth/host/hci_core.c:3803]: In hindsight probably yes, the bug got caught in a unittest though. If a 2nd request for splitting it out comes in I'll put it in a separate commit, otherwise I'll leave it as is

Comment (Thalley) [reply subsys/bluetooth/host/hci_core.c:3803]: I would agree with Johan that it's best to split it. 

I won't block this PR based on that though

Comment (kruithofa) [reply subsys/bluetooth/host/hci_core.c:3803]: I'll split

**Review by Thalley — APPROVED** [review]

**Review by sean-madigan — APPROVED** [review]

**Review by ppryga-nordic — APPROVED** [review]
